### ISSUE TITLE
FAILED to convert to mode STREAM_EXTERNAL_STATE_INFERENCE, tflite: Negative dimension size caused by subtracting 2 from 1

### ISSUE THREAD

**Issue by braveheartwithlove** [google-research/google-research#1208]
@rybakov  I need a help from you.
When I set the preprocess = 'mfcc', I could not convert the model to streaming mode correctly.  tf version=2.7.0.  Do you know the reason for this?
Thanks!

The error message is like:

W0720 23:15:13.229021 22708 stream.py:258] There is no need to use Stream on time dim with size 1
W0720 23:15:13.579084 22708 stream.py:258] There is no need to use Stream on time dim with size 1
W0720 23:15:13.660865 22708 test.py:646] FAILED to convert to mode STREAM_EXTERNAL_STATE_INFERENCE, tflite: Negative dimension size caused by subtracting 2 from 1 for '{{node streaming/average_pooling2d/AvgPool}} = AvgPool[T=DT_FLOAT, data_format="NHWC", ksize=[1, 2, 1, 1], padding="VALID", strides=[1, 2, 1, 1]](streaming/dropout_1/Identity)' with input shapes: [1,1,1,64].
2022-07-20 23:15:13.662452: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1850] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
I0720 23:15:13.663857 22708 test.py:449] tflite stream model state external with reset_state 1
I0720 23:15:33.224785 22708 model_train_eval.py:291] FAILED to run TFLite streaming: Model provided has model identifier '}☻

**Comment by braveheartwithlove** [comment]
The command I used:
python -m kws_streaming.train.model_train_eval 
--data_dir ./data/ 
--train_dir ./models/ds_tc_resnet/ 
--mel_upper_edge_hertz 7600 
--feature_type "mfcc_tf" 
--preprocess "mfcc"  
--causal_data_frame_padding 0 
--how_many_training_steps 1000,1000,500,500   
--learning_rate 0.001,0.0005,0.0002,0.0001 
--window_size_ms 30.0 
--window_stride_ms 10.0 
--mel_num_bins 40 
--dct_num_features 40 
--resample 0.15 
--alsologtostderr 
--train 1 
--time_shift_ms 100 
--use_spec_augment 1 
--time_masks_number 2 
--time_mask_max_size 25 
--frequency_masks_number 2 
--frequency_mask_max_size 7 
--pick_deterministically 1 
--unknown_percentage 200 
--silence_percentage 20 
--clip_duration_ms 1000 
--wanted_words yes 
ds_tc_resnet 
--dropout 0.0 
--ds_filters 128,64,64,64,128,128 
--ds_filter_separable 1,1,1,1,1,1 
--ds_repeat 1,1,1,1,1,1 
--ds_residual 0,1,1,1,0,0 
--ds_kernel_size 11,5,15,17,15,1 
--ds_stride 1,1,1,1,1,1 
--ds_dilation 1,1,1,1,2,1 
--ds_pool 1,2,1,1,1,1 
--activation "relu" 
--ds_padding "'causal','causal','causal','causal','causal','causal'"

**Comment by rybakov** [comment]
@braveheartwithlove "--preprocess "mfcc"" option uses [custom audio ops](https://github.com/google-research/google-research/blob/master/kws_streaming/layers/speech_features.py#L225) which is non streamable, so if you need to use it then you have to decopule you model from preprocessor, so that you will manage preprocessing on your own and stream the rest of the model which will take mfcc features as input (you will need to modify the model inputs). I would suggest not to use any preprocessing and feed raw audio, this way you can train and stream your model end to end: "--preprocess "raw""
